cmd/namespace-lock.go

```diff
@@ -295,6 +304,14 @@ func (li *lockInstance) GetLock(timeout *dynamicTimeout) (timedOutErr error) {
 
 // Unlock - block until write lock is released.
 func (li *lockInstance) Unlock() {
+	if li.ns.isDistXL {
+		if li.drwMutex == nil {
+			logger.LogIf(context.Background(), errUnexpected)
+			return
```

Comment:
Returning from Unlock() allows the call to proceed, we can never do that... while this may not happen. We should allow the server to crash when drwMutex is nil. So this kind of code is not needed.